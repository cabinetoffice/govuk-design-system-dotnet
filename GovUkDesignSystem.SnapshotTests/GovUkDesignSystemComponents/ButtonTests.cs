using System.Collections.Generic;
using System.Threading.Tasks;
using GovUkDesignSystem.GovUkDesignSystemComponents;
using GovUkDesignSystem.SnapshotTests.Helpers;
using Xunit;

namespace GovUkDesignSystem.SnapshotTests.GovUkDesignSystemComponents
{
    public class ButtonTests : SnapshotTestBase
    {
        private ButtonViewModel DefaultLabelViewModel()
        {
            return new ButtonViewModel
            {
                Name = "test-name",
                Value = "test-value",
                Disabled = false,
                PreventDoubleClick = true,
                PreventDoubleClickTimeout = 2,
                IsStartButton = false,
                Classes = "test-css-class",
                Attributes = new Dictionary<string, string> { { "attr-name", "attr-value" } },
                Text = "test text",
                Html = o => "test html"
            };
        }

        [Fact]
        public async Task Render_AllValues()
        {
            // Act & Assert
            await VerifyPartial("Button", DefaultLabelViewModel());
        }

        [Fact]
        public async Task Render_Disabled()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.Disabled = true;

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }

        [Fact]
        public async Task Render_StartButton()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.IsStartButton = true;

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }

        [Fact]
        public async Task Render_AsLink()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.Href = "test-href";

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }

        [Fact]
        public async Task Render_AsDisabledLink()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.Href = "test-href";
            viewModel.Disabled = true;

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }

        [Fact]
        public async Task Render_AsInput()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.Element = "input";

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }

        [Fact]
        public async Task Render_AsDisabledInput()
        {
            // Arrange
            var viewModel = DefaultLabelViewModel();
            viewModel.Element = "input";
            viewModel.Disabled = true;

            // Act & Assert
            await VerifyPartial("Button", viewModel);
        }
    }
}
