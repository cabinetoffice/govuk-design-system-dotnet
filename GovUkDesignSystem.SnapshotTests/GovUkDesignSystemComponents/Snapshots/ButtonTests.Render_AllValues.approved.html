<button class="govuk-button test-css-class" data-module="govuk-button" value="test-value" type="submit" name="test-name" aria-disabled="false" data-prevent-double-click="true" data-prevent-double-click-timeout="2" attr-name="attr-value">

test html        </button>